### PR TITLE
BED-4594+BED-4686: Styling fixes for `ExploreSearch` and `GraphButtons` components on the Explore page

### PR DESCRIPTION
**Pull request by elikmiller**
## Description

- Fixes an issue causing the "Search Current Results" input and result list to render behind the list of saved cypher queries when both are visible
- Fixes an issue causing the cypher query input to incorrectly overflow the available vertical space is exceeded
- Fixes an issue causing cypher query input linter messages to be cutoff when extending beyond the input box border

This is accomplished by placing the `ExploreSearch` and the `GraphButtons` components in the same vertical flexbox container and dynamically changing the size used by the cypher query input between 4 and 12 lines based on the available vertical viewport space. 

In order to co-locate these components (specifically the GraphButtons component) we make use of React's `useImperativeHandle` hook to enable ourselves to call functions which manipulate the Sigma.js graph instance from outside of the graph context. To learn more about this pattern please read https://react.dev/reference/react/useImperativeHandle#exposing-your-own-imperative-methods

## Motivation and Context

This PR addresses:
- BED-4594
- BED-4686

## How Has This Been Tested?

Unit tests have been updated

## Screenshots (optional):

https://github.com/user-attachments/assets/3f4d22c9-1f53-441f-ac6a-e221910aeeba

## Types of changes

- Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] I have met the contributing prerequisites
  - Assigned myself to this PR
  - Added the appropriate labels
  - Associated an issue: https://github.com/SpecterOps/BloodHound/issues/672
  - Read the Contributing guide: https://github.com/SpecterOps/BloodHound/wiki/Contributing
- [x] I have ensured that related documentation is up-to-date
  - Open API docs
  - Code comments (GoDocs / JSDocs)
- [x] I have followed proper test practices
  - Added/updated tests to cover my changes
  - All new and existing tests passed

### CODE REVIEW

**Review by StephenHinck — APPROVED** [review]
Functionally works great - I'll let others review the code itself for any issues there!

File: cmd/ui/src/views/Explore/ExploreSearch/CypherSearch.tsx
Comment (specter-flq) [inline cmd/ui/src/views/Explore/ExploreSearch/CypherSearch.tsx:59]: Not sure if its me, but I tested in chrome, and with overflow: 'scroll' it seems to be showing empty scrollbars even if no need to scroll (See screenshots below). But not in safari. If you change it to overflow: 'auto' it doesnt happen.

Screenshot 2024-09-10 at 3 20 12 PM
Screenshot 2024-09-10 at 3 21 23 PM

File: cmd/ui/src/views/Explore/ExploreSearch/CypherSearch.tsx
Comment (specter-flq) [inline cmd/ui/src/views/Explore/ExploreSearch/CypherSearch.tsx:77]: Same as above.

File: cmd/ui/src/views/Explore/ExploreSearch/CypherSearch.tsx
Comment (specter-flq) [inline cmd/ui/src/views/Explore/ExploreSearch/CypherSearch.tsx:170]: Not a huge deal but the overflow hidden attribute is making the buttons cut off in light mode. Doesnt seem like we need it though, but I will defer to you. 

Here is a screenshot with the overflow=hidden as implemented and below it one without the attribute:

Screenshot 2024-09-10 at 4 18 13 PM
Screenshot 2024-09-10 at 4 18 51 PM

**Review by specter-flq — CHANGES_REQUESTED** [review]
Code looks good, just two ui details I saw, not sure if its just me, but thought one of them was worth looking at before merging.

**Review by specter-flq — APPROVED** [review]
After talking with @urangel we decided to add the suggested changes to merge this in

**Review by urangel — APPROVED** [review]
🚀

**Pull request merged**
